src/journal/journald-server.c

```diff
@@ -1175,7 +1180,10 @@ void server_dispatch_message(
                                       NULL);
 
 finish:
-        dispatch_message_real(s, iovec, n, m, ucred, tv, label, label_len, unit_id, priority, object_pid);
+        /* restore cgroup path for logging */
+        if (c)
+                *c = '/';
+        dispatch_message_real(s, iovec, n, m, ucred, tv, label, label_len, unit_id, priority, object_pid, path);
```

Comment:
I was thinking about what happens if `path` contains two components, and the code above didn't find a third `/'. But indeed, then we'd never truncate `path` in the first place, and line 1185 would just overwrite an existing `/` with another `/`, so should be a no-op. So this should be fine after all.